x-pack/solutions/observability/plugins/apm/public/components/app/mobile/service_overview/index.tsx

```diff
@@ -35,9 +36,10 @@ import { useAdHocApmDataView } from '../../../../hooks/use_adhoc_apm_data_view';
 export const chartHeight = 288;
 
 export function MobileServiceOverview() {
-  const { serviceName } = useApmServiceContext();
+  const { serviceName, agentName } = useApmServiceContext();
   const router = useApmRouter();
   const { dataView } = useAdHocApmDataView();
+  const isAndroidAgent = isAndroidAgentName(agentName);
```

Comment:
Since this data will always be coming from otel, would it make more sense to use the `telemetry.sdk` fields? The `agent.name` gets constructed out of those fields anyway https://github.com/smith/opentelemetry-lib/blob/1a05d8fc8a7e7e0663e11419e0e40167a87ea54e/enrichments/trace/internal/elastic/resource.go#L78-L103

Reply:
I guess we could use `telemetry.sdk.name`, based on the available fields:

<img width="229" alt="Screenshot 2025-03-13 at 08 39 50" src="https://github.com/user-attachments/assets/dd29cc93-4c60-494b-b381-0542bf24ef38" />

However, those aren't available on the docs coming from the APM Server, so the dashboard would still show the broken components when using the tech-preview option.

Leaving the legacy option aside, I'm also not sure if we're comfortable relying on such generic values (like `android` in this case) to make UI decisions for our agents. Probably in the future we might want to come up with a more specific name for them to make sure we won't affect people using generic sdks. cc @AlexanderWert.